### PR TITLE
fix https://github.com/jhipster/generator-jhipster/issues/12471

### PR DESCRIPTION
**Pull request by thri33**
Removed unused validators imports

### CODE REVIEW

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 1]
```diff
@@ -3,7 +3,36 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
   import { mixins } from 'vue-class-component';
   import JhiDataUtils from '@/shared/data/data-utils.service';
 <% } %>
-import { numeric, required, minLength, maxLength, minValue, maxValue } from 'vuelidate/lib/validators';
+
```
Comment (qmonmert) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:6]: delete this empty line 😇

**Comment by pascalgrimaud** [comment]
@qmonmert : can you help on this one plz ?

**Comment by pascalgrimaud** [comment]
Ok forget my ask @qmonmert : I'm reviewing this, I think I got the fix to help here

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 2]
```diff
@@ -3,7 +3,35 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
   import { mixins } from 'vue-class-component';
   import JhiDataUtils from '@/shared/data/data-utils.service';
 <% } %>
-import { numeric, required, minLength, maxLength, minValue, maxValue } from 'vuelidate/lib/validators';
+<%_
+  let fieldsContainedValidators = [];
+  for (idx in fields) {
+    if (fields[idx].fieldValidate === true) {
+      if (!fieldsContainedValidators.includes('numeric') && fields[idx].fieldValidateRules.includes('numeric')) {
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:10]: this condition should be replaced by:

```
if (
        !fieldsContainedValidators.includes('numeric') && (
          fields[idx].fieldValidateRules.includes('numeric') || ['Integer', 'Long', 'Float', 'Double', 'BigDecimal'].includes(fields[idx].fieldType)
        )
      ) {
```

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 3]
```diff
@@ -3,7 +3,35 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
...
+      if (!fieldsContainedValidators.includes('required') && fields[idx].fieldValidateRules.includes('required')) {
+        fieldsContainedValidators.push('required');
+      }
+      if (!fieldsContainedValidators.includes('minLength') && fields[idx].fieldValidateRules.includes('minlength')) {
+        fieldsContainedValidators.push('minLength');
+      }
+      if (!fieldsContainedValidators.includes('maxLength') && fields[idx].fieldValidateRules.includes('maxlength')) {
+        fieldsContainedValidators.push('maxLength');
+      }
+      if (!fieldsContainedValidators.includes('minValue') && fields[idx].fieldValidateRules.includes('minValue')) {
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:22]: `fields[idx].fieldValidateRules.includes('minValue')` -> `fields[idx].fieldValidateRules.includes('min')`

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 4]
```diff
@@ -3,7 +3,35 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
...
+        fieldsContainedValidators.push('minValue');
+      }
+      if (!fieldsContainedValidators.includes('maxValue') && fields[idx].fieldValidateRules.includes('maxValue')) {
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:25]: `fields[idx].fieldValidateRules.includes('maxValue')` -> `fields[idx].fieldValidateRules.includes('max')`

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
@thri33 :
- can you add `fix https://github.com/jhipster/generator-jhipster/issues/12471` in description plz ? So the ticket will be closed automatically
- then, I added some reviews code, can you take care of it plz ?

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 5]
```diff
@@ -3,7 +3,35 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
...
+        fieldsContainedValidators.push('maxLength');
+      }
+      if (!fieldsContainedValidators.includes('minValue') && fields[idx].fieldValidateRules.includes('min')) {
+        fieldsContainedValidators.push('minValue');
+      }
+      if (!fieldsContainedValidators.includes('maxValue') && fields[idx].fieldValidateRules.includes('max')) {
+        fieldsContainedValidators.push('maxValue');
+      }
+    }
+  }
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:29]: just after that, I think we should add:

```javascript
  for (idx in relationships) {
    if (relationships[idx].relationshipValidate === true) {
      if (!fieldsContainedValidators.includes('required')) {
        fieldsContainedValidators.push('required');
      }
    }
  }
```

File: generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs [hunk 6]
```diff
@@ -3,7 +3,35 @@ import { Component,<% if (!fieldsContainBlob) { %> Vue,<% } %> Inject } from 'vu
...
+  if (fieldsContainedValidators.length !== 0) {
+    let validators = fieldsContainedValidators.toString();
+_%>
+    import { <%= validators %> } from 'vuelidate/lib/validators';
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/vue/src/main/webapp/app/entities/entity-update.component.ts.ejs:33]: remove the space before import

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @thri33

**Pull request merged**